File: lib/src/main/java/com/sd/lib/http/IPostRequest.java
package com.sd.lib.http;

import java.io.File;

public interface IPostRequest extends IRequest, IBodyRequest
{
    void addPart(String name, File file);

    void addPart(String name, String filename, String contentType, File file);

    void setParamsType(ParamsType type);

    enum ParamsType
    {
        Default,
        Json
    }
}

File: lib/src/main/java/com/sd/lib/http/IPostRequest.kt
package com.sd.lib.http

import java.io.File

interface IPostRequest : IRequest, IBodyRequest {

    fun addPart(name: String, file: File)

    fun addPart(name: String, file: File, filename: String? = null, contentType: String? = null)

    fun setParamsType(type: ParamsType)

    enum class ParamsType {
        Default, Json
    }
}